### PR TITLE
Update keyDrop function to not allow stupidness.

### LINKED ISSUE #372: Can drop car keys whilst in locked vehicle
<Please review the guidelines for contributing to this repository. The link is above.>
<If you are reporting a bug then you must use the provided issue template.>

### Expected behaviour
Should not allow players to drop the key to the vehicle they are in whilst it is locked as if they do they will become trapped and can only get out if a cop pulls them out or if they press escape and click respawn.

### Actual behaviour
Can trap yourself in your own car

### Steps to reproduce the behaviour 
Get in car, Lock car, Drop key, cry.

----

Mission version: 5.0
Branch: dev

### PR DESCRIPTION
**Pull request by SimZor**
<Please review the guidelines for contributing to this repository. The link is above.>

Resolves #372 

#### Changes proposed in this pull request: 
- Optimisation of function
- Do not allow a player to remove a key to vehicle if he/she is inside it

### CODE REVIEW

File: Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf [hunk 1]
```diff
@@ -3,24 +3,42 @@
...
+
+// Do not let them drop the key to a house
+if (_vehicle isKindOf "House_F") exitWith {
+    hint localize "STR_NOTF_cannotRemoveHouseKeys"
+};
+
+// Solve stupidness
+private _op = objectParent player;
+if (_op isEqualTo _vehicle && {locked _vehicle isEqualTo 2}) exitWith {
+    hint "You cannot drop keys to a vehicle which is locked and you are inside of."
```
Comment (Jawshy) [inline Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf]: @SimZor, could you please [localise](https://community.bistudio.com/wiki/localize) the above string [under notifications in the stringtable](https://github.com/AsYetUntitled/Framework/blob/b525f34ad9fe5435b61649cb1f2a086efebad3d4/Altis_Life.Altis/stringtable.xml#L3102)?

**Comment by SimZor** [comment]
@Jawshy

File: Altis_Life.Altis/stringtable.xml [hunk 2]
```diff
@@ -911,6 +911,17 @@
...
+        <Key ID="STR_NOTF_cannotDropKeys">
+            <Original>You cannot drop keys to a vehicle which is locked and you are inside of.</Original>
+            <Czech>You cannot drop keys to a vehicle which is locked and you are inside of.</Czech>
+            <French>You cannot drop keys to a vehicle which is locked and you are inside of.</French>
+            <Spanish>You cannot drop keys to a vehicle which is locked and you are inside of.</Spanish>
+            <Italian>You cannot drop keys to a vehicle which is locked and you are inside of.</Italian>
+            <Polish>You cannot drop keys to a vehicle which is locked and you are inside of.</Polish>
+            <Portuguese>You cannot drop keys to a vehicle which is locked and you are inside of.</Portuguese>
+            <Russian>You cannot drop keys to a vehicle which is locked and you are inside of.</Russian>
+            <German>You cannot drop keys to a vehicle which is locked and you are inside of.</German>
```
Comment (WolfSkin1337) [inline Altis_Life.Altis/stringtable.xml]: for German:
<German>You cannot drop keys to a vehicle which is locked and you are inside of.</German>
<German>Du kannst den Schlüssel nicht wegschmeißen, weil das Auto verschlossen ist und du drin sitzt.</German>

Comment (SimZor) [reply Altis_Life.Altis/stringtable.xml]: Cheers mate, will get this committed in a bit.

File: Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf [hunk 3]
```diff
@@ -3,24 +3,42 @@
...
+if (_sel isEqualTo -1) exitWith {
+    hint localize "STR_NOTF_noDataSelected"
+};
+
+if (_list lbData _sel isEqualTo "") exitWith {
+    hint localize "STR_NOTF_didNotSelectVehicle"
+};
+
+private _index = parseNumber (_list lbData _sel);
+private _vehicle = life_vehicles param [_index, objNull, [objNull]];
```
Comment (BoGuu) [inline Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf]: We can exit if NULL here

**Comment by SimZor** [comment]
Ready to merge? @BoGuu @Jawshy

File: Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf [hunk 4]
```diff
@@ -3,24 +3,43 @@
...
+if isNull _vehicle exitWith {};
+
+// Do not let them drop the key to a house
+if (_vehicle isKindOf "House_F") exitWith {
+    hint localize "STR_NOTF_cannotRemoveHouseKeys"
+};
+
+// Solve stupidness
+private _op = objectParent player;
```
Comment (BoGuu) [inline Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf]: We can avoid storing this one, can't see it being used below

File: Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf [hunk 5]
```diff
@@ -3,24 +3,43 @@
...
+
+// Update vehicle owners
+private _owners = _vehicle getVariable ["vehicle_info_owners", []];
 _owners deleteAt _index;
-_vehicle setVariable ["vehicle_info_owners",_owners,true];
+_vehicle setVariable ["vehicle_info_owners", _owners, true];
 
-[] spawn life_fnc_keyMenu;
+// Reload
+[] spawn life_fnc_keyMenu
```
Comment (BoGuu) [inline Altis_Life.Altis/core/pmenu/fn_keyDrop.sqf]: Just doubled check the fn, will be preferable to just call this

**Pull request merged**